### PR TITLE
OPSWAT Filescan Sandbox integration

### PR DESCRIPTION
**Pull request by anikobartos**
Changed the old OPSWAT integration to the Filescan Sandbox one.
Added some test case and modified the Readme.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -242,20 +242,23 @@ Example::
...
 Example::
 
-    OpswatAPI(apikey, 'windows7')
+    OPSWATSandboxAPI('mykey')
 
-OPSWAT sandbox on MetaDefender Cloud. Please create an account on `OPSWAT portal`_ to receive a free MetaDefender Cloud apikey.
+OPSWAT Filescan Sandbox. You can use the Activation Key that you received 
+from your OPSWAT Sales Representative, and follow the instructions on the 
+`OPSWAT Licence Activation`_ page or you can create an API key on the 
+`OPSWAT Community Site`_ under API Key tab.
```
Comment (petya2164) [inline README.rst]: Let's make it more specific:
OPSWAT Community Site => OPSWAT Filescan Community Site

File: sandboxapi/opswat.py [hunk 2]
```diff
@@ -1,26 +1,35 @@
 from __future__ import print_function
 
+# import json
```
Comment (petya2164) [inline sandboxapi/opswat.py]: Remove that line, if we don't need the import

File: sandboxapi/opswat.py [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
-        self.api_url = "https://api.metadefender.com/v4"
-        self.profile = profile or 'windows7'
-        self.api_token = apikey
+        self.api_key = api_key
+        self.api_url = url
+        self.headers = {"X-Api-Key": self.api_key}
         self.verify_ssl = verify_ssl
 
-    def analyze(self, handle, filename):
+    # def analyze(self, handle, filename, password = None):
```
Comment (petya2164) [inline sandboxapi/opswat.py]: Remove that line

File: sandboxapi/opswat.py [hunk 4]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
+        url_suffix = "/api/scan/{flow_id}/report?{postfix}".format(
+            flow_id=item_id, postfix=postfix
+        )
 
-        # else we try JSON
-        response = self._request(
-            "/sandbox/{sandbox_id}".format(sandbox_id=item_id), headers=headers)
+        response = self._request(url_suffix, headers=self.headers)
+
+        if report_format == "html":
```
Comment (petya2164) [inline sandboxapi/opswat.py]: Let's move this check to the start of this function.

File: sandboxapi/opswat.py [hunk 5]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
-            'apikey': self.api_token,
-        }
+        filters = [
+            "filter=general",
+            "filter=finalVerdict",
+            "filter=allTags",
+            "filter=overallState",
+            "filter=taskReference",
+            "filter=subtaskReferences",
+            "filter=allSignalGroups",
```
Comment (petya2164) [inline sandboxapi/opswat.py:156]: We can add `filter=iocs`.

File: sandboxapi/opswat.py [hunk 6]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
-    
+        api_key = sys.argv.pop()
+        url = sys.argv.pop()
+
     else:
         usage()
 
-    # instantiate Opswat Sandbox API interface.
-    opswat = OpswatAPI(apikey, 'windows7')
+    # instantiate OPSWAT Filescan Sandbox API interface.
```
Comment (petya2164) [inline sandboxapi/opswat.py]: I would delete this comment, it does not add any info.

File: sandboxapi/opswat.py [hunk 7]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
-    print(opswat.report(sandbox_id))
+    print("Analysis complete. fetching report...")
+    print(opswat.report(flow_id))
 
 
 if __name__ == "__main__":
 
     def usage():
-        msg = "%s: apikey <submit <fh> | available | report <id> | analyze <fh>"
+        msg = "%s: <url> <api_key> <submit <fh> | available | report <id> | score <report> | analyze <fh>"
```
Comment (petya2164) [inline sandboxapi/opswat.py]: Maybe we should clarify these parameters:
url => filescan_url
fh => file_path
id => flow_id

File: sandboxapi/opswat.py [hunk 8]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
+        api_key = sys.argv.pop()
+        url = sys.argv.pop()
+
     else:
         usage()
 
-    # instantiate Opswat Sandbox API interface.
-    opswat = OpswatAPI(apikey, 'windows7')
+    # instantiate OPSWAT Filescan Sandbox API interface.
+    opswat = OPSWATSandboxAPI(api_key)
```
Comment (petya2164) [inline sandboxapi/opswat.py]: The url parameter is missing here.

File: sandboxapi/opswat.py [hunk 9]
```diff
@@ -125,103 +132,115 @@ def report(self, item_id, report_format="json"):
...
+
+    if arg is None:
```
Comment (petya2164) [inline sandboxapi/opswat.py]: Let's modify this if:
```
if arg is None and "available" not in cmd:
```

File: sandboxapi/opswat.py (hunk 7 above)
Comment (petya2164) [reply sandboxapi/opswat.py]: report => flow_id

**Comment by petya2164** [comment]
Let's rename `item_id` to `flow_id` everywhere!

**Review by petya2164 — CHANGES_REQUESTED** [review]
See my comments above

**Comment by anikobartos** [comment]
> Let's rename `item_id` to `flow_id` everywhere!

@petya2164 
I used item_id because in the parent class it is called "item_id" and also in those functions which I need to override.
The other integrations use the same method that I implemented. Eg: cuckoo (id), falcon (job_id), fireeye (file_id), vmray (sample_id or analysis_id), but everybody use item_id.

**Pull request merged**